File: src/main/java/ru/hasa/springbootapp/dao/UserDao.java
package ru.hasa.springbootapp.dao;

import ru.hasa.springbootapp.model.User;

import java.util.List;

public interface UserDao {

    User findByUsername(String username);

    User saveUser(User user);

    void removeUserById(Long id);

    List<User> getAllUsers();

    User showUser(Long id);

    User updateUser(User user);

    User findByMail(String mail);
}

File: src/main/kotlin/ru/hasa/springbootapp/dao/UserDao.kt
package ru.hasa.springbootapp.dao

import ru.hasa.springbootapp.model.User

interface UserDao {
    fun findByUsername(username: String): User?
    fun saveUser(user: User): User
    fun removeUserById(id: Long)
    fun getAllUsers(): List<User>
    fun showUser(id: Long): User?
    fun updateUser(user: User): User
    fun findByMail(mail: String): User?
}
